api/lib/opentelemetry/context.rb

```diff
@@ -4,28 +4,146 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
+require 'opentelemetry/context/propagation'
```

Comment:
I'm still trying to wrap my head around this whole thing but I'm a little surprised to see this dep because the otep author seems to indicate
```
The Propagator API knows about the Context API, but the Context API does not need to know about Propagation.
```

https://github.com/open-telemetry/oteps/pull/66/files#r359638018

Reply:
The `Context` API is actually separate. That require is for namespacing reasons, so that all propagation code nests under `OpenTelemetry::Context::Propagation`. Since `OpenTelemetry::Context` doesn't actually depend on anything in `OpenTelemetry::Context::Propagation`, we could consider moving propagation up to the top level, `OpenTelemetry::Propagation`. However, propagation does depend on context, so there is a relationship there.